File: examples/publisher.cpp
#include <boost/asio.hpp>
#include <farfler/network/network.hpp>
#include <farfler/network/types.hpp>
#include <iostream>
#include <thread>

using namespace farfler::network;

int main() {
  // Create a network node
  boost::asio::io_context io_context;
  Network(io_context, "node");
  std::thread t([&io_context]() { io_context.run(); });

  // Subscribe to the "temperature" topic
  Network::Subscribe("temperature", [](const double &message) {
    std::cout << "Published temperature: " << message << std::endl;
  });

  while (true) {
    // Publish the temperature every second
    Network::Publish("temperature", 22.5);
    std::this_thread::sleep_for(std::chrono::seconds(1));
  }
}

File: examples/publisher.py
import farfler_network as fn
import time

# Create a network node
io_context = fn.IOContext()
network = fn.Network(io_context, "node")

def subscriber_callback(message):
  print(f"Published temperature: {message}")

# Create a subscriber
network.subscribe_float("temperature", subscriber_callback)
print("Subscribed to temperature messages")

while True:
  # Publish a message
  network.publish_float("temperature", 22.5)
  time.sleep(1)
